pkgs/development/libraries/science/math/tiny-cuda-nn/default.nix

```diff
@@ -0,0 +1,156 @@
+{
+  stdenv,
+  cmake,
+  fetchFromGitHub,
+  lib,
+  ninja,
+  symlinkJoin,
+  which,
+  pythonPackages,
+  pythonSupport ? false,
+}:
+assert lib.asserts.assertMsg
+(pythonPackages.torch.cudaSupport)
+"tiny-cuda-nn requires torch to be built with cudaSupport"; let
+  inherit (lib) lists strings;
+  inherit (pythonPackages.torch) cudaPackages cudaSupport;
+  inherit (cudaPackages) cudatoolkit cudaFlags;
+
+  cuda-redist = symlinkJoin {
+    name = "cuda-redist";
+    paths = with cudaPackages; [
+      cuda_cudart
+      cuda_nvcc
+      libcublas
+      libcusparse
+      libcusolver
+    ];
+  };
+in
+  stdenv.mkDerivation (finalAttrs: {
```

Comment:
is there a reason to prefer this over just having separate derivations for the cpp build and the python package? seems like that might be a bit simpler?

Reply:
That's what I had done originally because I couldn't figure out how `toPythonModule` was supposed to work. From the docs, I got the impression that libraries or applications which provide python bindings should use the same derivation and apply `toPythonModule` or the like in `python-packages.nix` to create the python version.

I'm open to going back to that, though the non-standard location of the torch bindings also complicated the Python derivation IIRC, so I'm not sure how much cleaner it would be.

Do you have an example of an application which does this? I'd like to make sure they use the same source so there's no drift between the C++ library and torch bindings.

Reply:
tbh i haven't done much work on packages that are half python/half cpp... so i'm not sure what the recommended pattern is. It just struck me as a bit more complex this way. But if this is the recommendation I can't object to that. Do you have a link to the docs you're referring to?

Reply:
https://nixos.org/manual/nixpkgs/unstable/#topythonmodule-function